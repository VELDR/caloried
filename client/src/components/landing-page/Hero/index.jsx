import { FormattedMessage } from 'react-intl';
import { useNavigate } from 'react-router-dom';
import { East } from '@mui/icons-material';

import classes from './style.module.scss';

const Hero = () => {
  const navigate = useNavigate();
  return (
    <div className={classes.heroContainer}>
      <div className={classes.hero}>
        <div className={classes.heroTitle}>
          <FormattedMessage id="app_hero_title" />
        </div>
        <div className={classes.heroSubtitle}>
          <FormattedMessage id="app_hero_subtitle" />
        </div>
        <div className={classes.heroDescription}>
          <FormattedMessage id="app_hero_description" />
        </div>
        <div className={classes.heroButton} onClick={() => navigate('/sign-up')}>
          <span>
            <FormattedMessage id="app_hero_button" />
          </span>
          <East className={classes.buttonArrow} />
        </div>
      </div>
    </div>
  );
};

export default Hero;
