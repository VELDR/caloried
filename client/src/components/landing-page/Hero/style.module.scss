@import '@styles/base';

.heroContainer {
  background-image: linear-gradient(rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5)), url('@static/images/landingpage.png');
  background-repeat: no-repeat;
  background-position: bottom;
  background-size: cover;
  width: 100%;
  height: 25rem;
  @include apply-at-md-size {
    height: 100vh;
  }
  .hero {
    @include flex-center-column;
    gap: 0.5rem;
    margin: 0 auto;
    padding: 0 1.5rem;
    height: 100%;
    text-align: center;
    color: $color-bg-primary;
    @include apply-at-md-size {
      max-width: 40rem;
    }
    .heroTitle {
      font-size: $font-size-huge;
      font-weight: 700;
      background-image: linear-gradient(to bottom, var(--color-primary), var(--color-accent));
      color: transparent;
      background-clip: text;
      -webkit-background-clip: text;
      -webkit-text-fill-color: transparent;

      @include apply-at-md-size {
        font-size: 3rem;
      }
    }
    .heroSubtitle {
      font-size: $font-size-large;
      font-weight: 500;
      @include apply-at-md-size {
        font-size: $font-size-larger;
      }
    }
    .heroDescription {
      margin-top: 1rem;
      font-size: $font-size-regular;
      max-width: 15rem;
      @include apply-at-md-size {
        width: 20rem;
        font-size: $font-size-regular;
      }
    }
    .heroButton {
      @include flex-center;
      gap: 0.35rem;
      cursor: pointer;
      position: relative;
      background: linear-gradient(to bottom, var(--color-primary), var(--color-accent));
      padding: 0.25rem 0.75rem;
      border-radius: 0.5rem;
      transition: all 0.2s ease;
      margin-top: 2rem;
      font-weight: 500;
      &:hover {
        box-shadow: 0 0 10px 1px $color-accent;
      }
      .buttonArrow {
        @include box(1rem);
        @include apply-at-md-size {
          @include box(1.1rem);
        }
      }
    }
  }
}
