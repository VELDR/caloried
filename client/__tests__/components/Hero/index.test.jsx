import Hero from '@components/landing-page/Hero';
import { fireEvent, render } from '@utils/testHelper';

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  useNavigate: () => mockNavigate,
}));

let wrapper;

beforeEach(() => {
  wrapper = render(<Hero />);
});

describe('Hero Component', () => {
  it('should render the component correctly', () => {
    const { getByTestId } = wrapper;

    expect(getByTestId('hero')).toBeInTheDocument();
  });

  it('should call navigate to sign up when hero button is clicked', () => {
    const { getByTestId } = wrapper;

    const heroButton = getByTestId('hero-button');
    fireEvent.click(heroButton);

    expect(mockNavigate).toHaveBeenCalledWith('/sign-up');
  });

  it('should match with snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
