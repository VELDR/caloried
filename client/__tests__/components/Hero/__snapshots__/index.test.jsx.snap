// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Component should match with snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="heroContainer"
        data-testid="hero"
      >
        <div
          class="hero"
        >
          <div
            class="heroTitle"
          >
            Makan Cerdas, Hidup Sehat
          </div>
          <div
            class="heroSubtitle"
          >
            Panduan Nutrisi Pribadi Anda
          </div>
          <div
            class="heroDescription"
          >
            Caloried memudahkan pemantauan dan penyusunan diet Anda
          </div>
          <div
            class="heroButton"
            data-testid="hero-button"
          >
            <span>
              Mulai Sekarang
            </span>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium buttonArrow css-i4bv87-MuiSvgIcon-root"
              data-testid="EastIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m15 5-1.41 1.41L18.17 11H2v2h16.17l-4.59 4.59L15 19l7-7-7-7z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="heroContainer"
      data-testid="hero"
    >
      <div
        class="hero"
      >
        <div
          class="heroTitle"
        >
          Makan Cerdas, Hidup Sehat
        </div>
        <div
          class="heroSubtitle"
        >
          Panduan Nutrisi Pribadi Anda
        </div>
        <div
          class="heroDescription"
        >
          Caloried memudahkan pemantauan dan penyusunan diet Anda
        </div>
        <div
          class="heroButton"
          data-testid="hero-button"
        >
          <span>
            Mulai Sekarang
          </span>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium buttonArrow css-i4bv87-MuiSvgIcon-root"
            data-testid="EastIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m15 5-1.41 1.41L18.17 11H2v2h16.17l-4.59 4.59L15 19l7-7-7-7z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
